components/space/Habits/HabitsOptions/edit-habit.tsx

```diff
@@ -0,0 +1,144 @@
+'use client';
+import { FormikStep, FormikStepper } from '@/components/shared/FormikStepper/formik-stepper';
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+import { Field, FormikValues } from 'formik';
+import React, { useState } from 'react';
+import { number, object, string } from 'yup';
+import { patchHabit } from '@/lib/actions';
+import { HabitT, timeFrames } from '@/lib/types-and-constants';
+import { TextField as TextFieldFormikMui } from 'formik-mui';
+import { MenuItem } from '@mui/material';
+
+interface EditHabitProps {
+    habit: HabitT;
+    handleCloseDialog?: () => void;
+}
+
+/**
+ * Component using the FormikStepper to edit the description of the habit
+ * @param habit
+ * @returns
+ */
+export function EditHabit({ habit, handleCloseDialog }: EditHabitProps) {
+    const [habitDescripton, setHabitDescripton] = useState<string | undefined>(habit.description);
+
+    const [habitTimes, setHabitTimes] = useState<number>(habit.times);
+    const [habitTimeFrame, setHabitTimeFrame] = useState<string>(habit.time_frame);
+
+    /**
+     * Submits the request to the server action which patches the habit
+     */
+    async function submitEditHabit(values: FormikValues, habitId: number) {
+        const newHabit = {
+            title: habit.title,
+            description: values.description,
+            times: values.times,
+            time_frame: values.time_frame,
+            spaces: habit.spaces,
+        };
+        const updatedHabit: HabitT = await patchHabit(newHabit, habitId);
```

Comment:
all this should not be necessary (creating a complete new habit) if we use the PATCH method instead of the PUT
1. check were the method patchHabit() is being used
2. Edit where necessary (no need to set all the fields again)
3. Test
4. Is bug still present? : "when editing habit modal, you can not change the description and the times per week at the same time, the description automatically changes to the original when you change times per week (but no vice -versa)"